src/System.Management.Automation/namespaces/RegistryProvider.cs

```diff
@@ -1825,8 +1825,19 @@ public void GetProperty(
                     notePropertyName = LocalizedDefaultToken;
                 }
 
-                propertyResults.Properties.Add(new PSNoteProperty(notePropertyName, key.GetValue(valueName)));
-                valueAdded = true;
+                try
+                {
+                    propertyResults.Properties.Add(new PSNoteProperty(notePropertyName, key.GetValue(valueName)));
+                    valueAdded = true;
+                }
+                catch (InvalidCastException invalidCast)
+                {
+                    WriteError(new ErrorRecord(
+                        invalidCast,
+                        invalidCast.GetType().FullName,
```

Comment:
I suggest to use "InvalidCast" const - it is our common pattern.

Reply:
@iSazonov Do you mean replacing the Error ID with a constant?

Reply:
Yes, you can find many such patterns in code.

Reply:
I found a pattern you mentioned. I agree to follow existing pattern.

#Closed